### PR TITLE
podmanv2 commit

### PR DESCRIPTION
**Pull request by baude**
add commit of a container to a container-image

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: cmd/podmanV2/containers/commit.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	iidFile string
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: commitCommand,
+	})
+	flags := commitCommand.Flags()
+	flags.StringArrayVarP(&commitOptions.Changes, "change", "c", []string{}, fmt.Sprintf("Apply the following possible instructions to the created image (default []): %s", strings.Join(ChangeCmds, " | ")))
```
Comment (jwhonce) [inline cmd/podmanV2/containers/commit.go]:
```suggestion
	flags.StringArrayVarP(&commitOptions.Changes, "change", "c", []string{}, "Apply the following possible instructions to the created image (default []): " + strings.Join(ChangeCmds, " | "))
```
Is faster and uses less resources.

File: cmd/podmanV2/containers/commit.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	commitDescription = `Create an image from a container's changes. Optionally tag the image created, set the author with the --author flag, set the commit message with the --message flag, and make changes to the instructions with the --change flag.`
+
+	commitCommand = &cobra.Command{
+		Use:     "commit [flags] CONTAINER [IMAGE]",
+		Short:   "Create new image based on the changed container",
```
Comment (jwhonce) [inline cmd/podmanV2/containers/commit.go]:
```suggestion
		Short:   "Create new image based on the changed container",
		Args:    cobra.MinimumNArgs(1),
```

File: cmd/podmanV2/containers/commit.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	flags.StringVarP(&commitOptions.Author, "author", "a", "", "Set the author for the image committed")
+	flags.BoolVarP(&commitOptions.Pause, "pause", "p", false, "Pause container during commit")
+	flags.BoolVarP(&commitOptions.Quiet, "quiet", "q", false, "Suppress output")
+	flags.BoolVar(&commitOptions.IncludeVolumes, "include-volumes", false, "Include container volumes as image volumes")
+
+}
+func commit(cmd *cobra.Command, args []string) error {
+	if len(args) < 1 {
+		return errors.Errorf("you must provide a container name or ID and optionally a target image name")
+	}
```
Comment (jwhonce) [inline cmd/podmanV2/containers/commit.go]:
```suggestion
```
Use cobra.MinimumNArgs(1) above

Comment (TomSweeneyRedHat) [reply cmd/podmanV2/containers/commit.go]: @jwhonce if you don't provide a container name, what does it return as an error if you use cobra.MinimunNArgs?

**Comment by baude** [comment]
fixed up

File: cmd/podmanV2/containers/commit.go (hunk 3 above)
Comment (jwhonce) [reply cmd/podmanV2/containers/commit.go]: The error message says you need an argument.

Comment (TomSweeneyRedHat) [reply cmd/podmanV2/containers/commit.go]: Personally I think that's an ease of use for the developer and a downgrade for the end user.  I'm not in favor.

Other than that, the code LGTM.

**Comment by baude** [comment]
merge me please

**Pull request merged**